backend/libinput/events.c

```diff
@@ -24,6 +24,11 @@ struct wlr_input_device *get_appropriate_device(
 }
 
 static void wlr_libinput_device_destroy(struct wlr_input_device_state *state) {
+	list_t *devices = libinput_device_get_user_data(state->handle);
+	// devices themselves are freed in wlr_libinput_backend_destroy
+	// this list only has a part of the same elements so just free list
+	list_free(devices);
```

Comment:
If we don't free devices here, the devices we create line 65 (added with libinput_device_set_user_data) is never freed.

Turns out that also causes a double-free on different hardware now I'm testing on a different machine, trying to figure why before it explodes for too many others... Ah, I see, the same devices (list) can be pointed to in multiple wlr_input_device_state if the device has multiple capabilities.
Not sure how to free exactly once without tracking all the lists at this point, will have to think it through - state->devices could be a list to all the devices list but it's not particularily elegant